libraries/ms-helper/src/relay_rx.c

```diff
@@ -0,0 +1,57 @@
+#include "relay_rx.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_unpack.h"
+#include "critical_section.h"
+#include "gpio.h"
+#include "status.h"
+
+// CANRxHandler
+static StatusCode prv_relay_rx_can_handler(const CANMessage *msg, void *context,
+                                           CANAckStatus *ack_reply) {
+  RelayRxStorage *storage = context;
+  uint8_t state = storage->state_bound;
+  // NOTE: This is a bit of a hack that exploits the fact all the relay control messages are the
+  // same. The aim here is to not necessarily force a constraint based on message id/name. Instead
+  // all single field u8 messages will succeed in unpacking but the contents must be valid.
+  CAN_UNPACK_BATTERY_RELAY(msg, &state);
+  if (state >= storage->state_bound) {
+    *ack_reply = CAN_ACK_STATUS_INVALID;
+  } else {
+    storage->curr_state = state;
```

Comment:
If the current state is only used for the handler, I assume its storage is just for debugging?

Reply:
Yeah. Also possibly if the device to keep track for some reason but I didn't expose it via a function since I didn't see a need?